x-pack/plugins/reporting/server/config/index.ts

```diff
@@ -21,7 +21,7 @@ export const config: PluginConfigDescriptor<ReportingConfigType> = {
     unused('capture.viewport', { level: 'warning' }), // deprecated as unused since 7.16
     (settings, fromPath, addDeprecation) => {
       const reporting = get(settings, fromPath);
-      if (reporting?.roles?.enabled !== false) {
+      if (reporting?.roles?.enabled !== false && reporting?.statefulSettings?.enabled) {
```

Comment:
Now that the proper values are in the serverless config, we don't need this change as the original logic is more fitting.